src/LayerEditor/ui/diagram_editor/diagram_scene.py

```diff
@@ -38,12 +40,23 @@ def __init__(self, block, parent):
         # polygons
         self.decomposition = block.decomposition
         res = self.decomposition.get_last_polygon_changes()
+        self.decomposition.set_tolerance(self.TOLERANCE)
         #assert res[0] == PolygonChange.add
         self.outer_id = res[1]
         """Decomposition of the a plane into polygons."""
         self.update_scene()
         self.pixmap_item = None
 
+    def get_shape(self, dim, shape_id) -> [GsPoint, GsSegment, GsPolygon]:
```

Comment:
This would mean creating `self.shape` instead of `self.points`, `self.segments` and `self.polygons` and replacing all (more than 30) occurrences. It might not be worth it, but I leave it up to you.

Reply:
`get_shape` was not really necessary and was removed.